### ISSUE TITLE
ConfigurationProperties.getConfigIndexForPlatform reads 'undefined' at C_Cpp_ConfigurationProperties.js:238:35

### ISSUE THREAD

**Issue by bobbrow** [microsoft/vscode-cpptools#989]
From @dcourtois:  (separating this out from #19)

Thanks for the answer. I works kind of randomly on my computer ... sometimes it works fine, and some other time, doesn't. When it doesn't, the developer console shows the following error:

```
Cannot read property 'configurations' of undefined: TypeError: Cannot read property 'configurations' of undefined
    at ConfigurationProperties.getConfigIndexForPlatform (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:238:35)
    at new ConfigurationProperties (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:149:51)
    at Object.setupConfigurationProperties (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:532:15)
    at Object.activate (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp.js:92:76)
    at util.readFileText.then.catch.then (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/main.js:119:19)
```

Don't know why, but that's pretty annoying :)

**Comment by bobbrow** [comment]
@dcourtois, do you see any other error output in the console?  I see a spot in the code where it's possible to hit this if the extension failed to parse your `${workspaceRoot}/.vscode/c_cpp_properties.json` file.  You might only see it when you first open the folder in VS Code.

**Comment by dcourtois** [comment]
Hi, sorry I left work, so I won't be able to give you more informations until next monday.
I'm generating the `c_cpp_properties.json` file with a modified version of CMake Tools extension (you commented on my pull request there, a few weeks agao :) ) so maybe this has something to do with this bug (if CppTools tries to read the json file while CMake Tools is actually writing it, or something like this ?)

**Comment by dcourtois** [comment]
Hi again, back to work. Here are the only errors that I have:

```
[Extension Host] rejected promise not handled within 1 second
t.logExtensionHostMessage @ /usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29
/usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29 Invalid backend promise: Error: Invalid backend promise
    at new CMakeToolsWrapper (/home/dcourtois/.vscode/extensions/vector-of-bool.cmake-tools-0.10.2/out/src/wrapper.js:46:40)
    at /home/dcourtois/.vscode/extensions/vector-of-bool.cmake-tools-0.10.2/out/src/extension.js:17:23
    at Generator.next (<anonymous>)
    at /home/dcourtois/.vscode/extensions/vector-of-bool.cmake-tools-0.10.2/out/src/extension.js:7:65
    at __awaiter (/home/dcourtois/.vscode/extensions/vector-of-bool.cmake-tools-0.10.2/out/src/extension.js:3:12)
    at activate (/home/dcourtois/.vscode/extensions/vector-of-bool.cmake-tools-0.10.2/out/src/extension.js:15:12)
    at Function.t._callActivateOptional (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:424654)
    at Function.t._callActivate (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:424439)
    at /usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:424214
    at Object.m [as _notify] (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:41150)
e.onUnexpectedError @ /usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29
/usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29 [Extension Host] rejected promise not handled within 1 second
t.logExtensionHostMessage @ /usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29
/usr/share/code/resources/app/out/vs/workbench/workbench.main.js:29 Cannot read property 'configurations' of undefined: TypeError: Cannot read property 'configurations' of undefined
    at ConfigurationProperties.getConfigIndexForPlatform (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:238:35)
    at new ConfigurationProperties (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:149:51)
    at Object.setupConfigurationProperties (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp_ConfigurationProperties.js:532:15)
    at Object.activate (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp.js:92:76)
    at util.readFileText.then.catch.then (/home/dcourtois/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/main.js:119:19)
```

The only other one I have is an issue with CMake Tools that I always had, and which never seemed to cause problems.

And once I have the error in cpptools, no matter what I do, I can't make it work again: manually updating the `c_cpp_properties.json` file won't do anything.

(at the moment I'm still trying to find in which circumstances it works, but with no success so far ...)

**Comment by bobbrow** [comment]
The fix for this should be in 0.12.4.  Let me know if you continue to see this problem.
